drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,394 @@
+/* ST Microelectronics LIS2DUX12 3-axis accelerometer driver
+ *
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2dux12.pdf
+ */
+
+#define DT_DRV_COMPAT st_lis2dux12
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "lis2dux12.h"
+
+LOG_MODULE_REGISTER(LIS2DUX12, CONFIG_SENSOR_LOG_LEVEL);
+
+static int lis2dux12_set_odr(const struct device *dev, uint8_t odr)
+{
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lis2dux12_md_t mode = {.odr = odr};
+
+	return lis2dux12_mode_set(ctx, &mode);
+}
+
+/**
+ * @brief  Accelerometer full-scale selection.[set]
+ *
+ * @param  ctx      read / write interface definitions
+ * @param  val      change the values of fs in reg CTRL5
+ * @retval          interface status (MANDATORY: return 0 -> no Error)
+ *
+ */
+static int32_t lis2dux12_full_scale_set(stmdev_ctx_t *ctx, uint8_t val)
```

Comment:
this function doesn't seem to be used, and do you really need `int32_t` for this?

Reply:
I wasn't sure whether to leave this or not. The reference implementation (`lis2dw12`) had a similar method and it is conceivable the accelerometer scale may need to be changed at runtime. The return value was propagated from `lis2dux12_read_reg`, which returns an `int32_t`.

Reply:
> The return value was propagated from `lis2dux12_read_reg`, which returns an `int32_t`.

hmm.. `lis2dw12_full_scale_set` also has `int32_t` but is stored in `int` in the driver

Reply:
I modified the `set_range` implementation to also call `lis2dux12_full_scale_set()` as well as set the gain.

Here is the equivalent method from the `lis2dw12`:

https://github.com/zephyrproject-rtos/zephyr/blob/e87ded3f03798b2f2f348984f673489542a127cc/drivers/sensor/lis2dw12/lis2dw12.c#L35-L56